src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private V nullKeyValue = null;
+    private boolean nullKeyPresent = false;
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this.table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            if (!nullKeyPresent) {
+                size++;
+            }
+            nullKeyValue = value;
+            nullKeyPresent = true;
+            return;
+        }
+        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = getBucketIndex(key);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+        }
+        addNode(key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return nullKeyPresent ? nullKeyValue : null;
+        }
+        int index = getBucketIndex(key);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void addNode(K key, V value, int index) {
+        Node<K, V> newNode = new Node<>(key, value, table[index]);
+        table[index] = newNode;
+        size++;
+    }
+
+    private void resize() {
+        int newCapacity = table.length * 2;
```

Comment:
```suggestion
        int newCapacity = table.length * GROW_FACTOR;
```
magic number - to a constant